### ISSUE TITLE
[FEATURE] Bump Elemental-Toolkit Elemental-CLI

### ISSUE THREAD

**Issue by tjjh89017** [harvester/harvester#2565]
**Is your feature request related to a problem? Please describe.**

Bump Elemental-Toolkit Elemental-CLI

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

- os2 related
    - Luet quay.io/costoolkit/releases-teal is not always the newest
        - need to pin reference
- harvester-installer related
    - harv-install needs changes
        - elemental-cli args are changed
    - installer needs to change the partition layout gen 
        - https://github.com/rancher/elemental-cli/blob/main/config.yaml.example
        - https://github.com/harvester/harvester-installer/blob/d547dbdfecb102a7582c45264e3b7623b4eb7e1c/pkg/config/cos.go#L533
- OEM
    - need to install 99_custom.yaml to /oem manually
    - or need some other way to let elemental-cli install oem.yaml for us
- Eelemental related
    - (?, not sure) Mount point is different
- Upgrade command
    - elemental upgrade has new parameter, we need to patch the upgrade script

**Comment by tjjh89017** [comment]
wait for elemental cli to support extra partition installation

**Comment by TachunLin** [comment]
Verified on Harvester with `master-f3475489-head` (23/01/10) for the following check point after bumped Elemental took-kit to 

* Can deploy 3 nodes Harvester cluster using ISO installation 
* Can deploy 3 nodes Harvester cluster using iPXE install with common config-create and config-join settings 
* The elemental version already bumped to `v0.1.0+gb918db3`
```
n1-230110:~ # elemental version
v0.1.0+gb918db3

n2-230110:~ # elemental version
v0.1.0+gb918db3

n3-230110:~ # elemental version
v0.1.0+gb918db3

**Comment by TachunLin** [comment]
Working on the verification of `os` type Harvester config on local bare machine cluster 
refer to https://docs.harvesterhci.io/v1.1/install/harvester-configuration/#oswrite_files

- [x] os.hostname 
- [x] os.password 
- [x] os.ntp_servers
- [x] os.ssh_authorized_keys
  ```
  davidtclin@localhost:/srv/www/htdocs> ssh rancher@192.168.50.101
  rancher@node1:~> grep "nameserver" /etc/resolv.conf
  nameserver 8.8.8.8
  nameserver 1.1.1.1
  nameserver 192.168.50.1
  ```
- [x] os.write_files
  ```
  node1:~ # cat /etc/connman/main.conf
  node1:~ # cat /etc/crontab
  15 * * * * root ship_logs
  node1:~ # ls /home/rancher/binary.gif
  /home/rancher/binary.gif
  node1:~ # ls -al /home/rancher/binary.gif
  -r-xr-xr-x 1 root root 373 Jan 13 08:45 /home/rancher/binary.gif
  ```
- [ ] os.modules
  ```
  node1:~ # lsmod | grep kvm
  kvm_intel             270336  0
  kvm                   794624  1 kvm_intel
  irqbypass              20480  1 kvm
  node1:~ # lsmod | grep nvme
  ```
- [x] os.sysctls
  ```
  node1:~ # sysctl -a | grep kernel.printk
  kernel.printk = 4	4	1	7
  kernel.printk_delay = 0
  kernel.printk_devkmsg = on
  kernel.printk_ratelimit = 5
  kernel.printk_ratelimit_burst = 10
  node1:~ # sysctl -a | grep kernel.kptr_restrict
  kernel.kptr_restrict = 1
  
  ```
- [x] os.dns_nameservers
- [x] os.environment
  ```
  node1:~ # cat /etc/environment
  http_proxy="http://192.168.50.76:3128"
  https_proxy="http://192.168.50.76:3128"
  ```
- [x] os.labels
![image](https://user-images.githubusercontent.com/29251855/212257789-16e6382b-694e-4145-9e7e-0680a8d9aa01.png)

**Comment by TachunLin** [comment]
Verified **fixed** on `master-f3475489-head` (23/1/10). Close this issue.

### Result
* We can use ISO and pxe installation to correctly provision Harvester and confirmed the elemental tool-kit have been bumped to desired version in result: 
https://github.com/harvester/harvester/issues/2565#issuecomment-1378491289

* And check most of the `os` harvester configuration setting can work on pxe installation in result:
https://github.com/harvester/harvester/issues/2565#issuecomment-1378494699

### Test Information
* Test Environment: 3 nodes harvester on bare machines
* Harvester version: `master-f3475489-head` (23/1/10)

### Verify Steps

1. New cluster deployment in ISO and ipxe installation.  
2. Login to nodes and verify the elemental version is v0.1.0+gb918db3 (by typing elemental version)
3. New cluster deployment (iPXE), this should cover most .install settings in https://docs.harvesterhci.io/v1.0/install/harvester-configuration/
4. Select some `os` settings and verify if it works. 


### Additional Context
We found that unable to load nvme module after using os.module Harvester config for installation
This did not related to elemental tool-kit bump.
We will track this issue in #3366
